datumaro/util/annotation_util.py

```diff
@@ -14,6 +14,16 @@
 from datumaro.util.mask_tools import mask_to_rle
 
 
+def __getattr__(name: str):
+    if name == 'smooth_line':
+        import warnings
+        warnings.warn(f"Using {name} from '{__package__}' is deprecated, "
+            "the function is renamed to 'approximate_line'", DeprecationWarning,
+            stacklevel=2)
+
+        return approximate_line
```

Comment:
From what I see in the diff, `approximate_line` doesn't return the same thing as `smooth_line` did (`new_points, step` vs `new_points`), so can it really be considered a compatible replacement?

Reply:
Maybe we should just remove the previous version or leave the previous one and mark it deprecated, while introducing a new one? I feel, it was not widely used, but I can't be sure, of course. I noticed that the step value was totally incorrect. It was supposed to be in the resulting line metric, but it was in the original curve metric and had no relation to the resulting one. And then I understood we should not return a single step at all, because it is not equal in the resulting curve.

Reply:
I [searched on GitHub](https://github.com/search?p=2&q=annotation_util+smooth_line&type=Code) and didn't find any uses that weren't in copies of Datumaro itself or CVAT. So I don't think it would be a big problem to remove it.

Reply:
Removed